RATapi/classlist.py

```diff
@@ -401,3 +403,59 @@ def _determine_class_handle(input_list: Sequence[object]):
             class_handle = type(input_list[0])
 
         return class_handle
+
+    # Pydantic core schema which allows ClassLists to be validated
+    # in short: it validates that each ClassList is indeed a ClassList,
+    # and then validates ClassList.data as though it were a typed list
+    # e.g. ClassList[str] data is validated like list[str]
+    @classmethod
+    def __get_pydantic_core_schema__(cls, source: Any, handler):
+        # import here so that the ClassList can be instantiated and used without Pydantic installed
+        from pydantic import ValidatorFunctionWrapHandler
+        from pydantic.types import (
+            core_schema,  # import core_schema through here rather than making pydantic_core a dependency
+        )
+        from typing_extensions import get_args, get_origin
+
+        # if annotated with a class, get the item type of that class
+        origin = get_origin(source)
+        if origin is None:
+            origin = source
+            item_tp = Any
+        else:
+            item_tp = get_args(source)[0]
+
+        item_schema = handler.generate_schema(list[item_tp])
+
+        def coerce(v: Any, handler: ValidatorFunctionWrapHandler) -> ClassList[T]:
+            """If a list is given, try to coerce it to a ClassList."""
+            if isinstance(v, list):
```

Comment:
ah i didn't know there was a `sequence_like` input! i originally set it up for any `Iterable` but it wasn't working great because strings are iterable. this should work better

Reply:
`sequence_like` is internal to pydantic so i might just go for `Sequence` instead. the side-effect here (unless I add a check for it) is that a string will be coerced to a ClassList[str] (this isn't too weird, as strings can be coerced to list[str] already). I can live with that if you think that's acceptable?

Reply:
We could go for "sequence and not str" as we do for the ClassList initialisation? The problem I always have with string conversion is that `"string"` ends up as `["s", "t", "r", "i", "n", "g"]`